ibm/service/metricsrouter/resource_ibm_metrics_router_settings.go

```diff
@@ -0,0 +1,339 @@
+// Copyright IBM Corp. 2023 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package metricsrouter
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/validate"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/metricsrouterv3"
+)
+
+func ResourceIBMMetricsRouterSettings() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMMetricsRouterSettingsCreate,
+		ReadContext:   resourceIBMMetricsRouterSettingsRead,
+		UpdateContext: resourceIBMMetricsRouterSettingsUpdate,
+		DeleteContext: resourceIBMMetricsRouterSettingsDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"default_targets": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
```

Comment:
All arguments are optional without any required

Reply:
yes all are optional
made primary_metadata_region as Computed and Optional both true as primary_metadata_region can't be made null once set.